### DISCUSSION TITLE
Switch to deactivate automatic gate closing while already implemented in the Gate logic

### DISCUSSION THREAD

**Discussion by djiwhy** [etiennec78/Home-Automation#2] [Ideas]
I have a sliding gate whose control logic already allows it to close automatically after 30 seconds.
The blueprint works very well but when it wants to close the gate, it sends a command that causes my gate to lock in place.

Is it possible to deactivate the blueprint part that manages closing?

To control a gate with a timed automatic closing function, the user would have to be able to deactivate the blueprint's automatic closing function.

Thank you to those who know how to do this, and who have already tried it, for sharing it with a poor ignoramus like me 😁

Translated with DeepL.com parce que mon anglais est pas super bon 😅

**Comment by etiennec78** [comment]
Hello !
I've pushed an update to add the feature you requested
Please keep in mind that disabling gate automatic closing will also disable any security after the gate has been open.

PS : T'inquiètes c'était assez clair pour que je comprenne 👍

**Comment by djiwhy** [comment]
Hi, nice reactivity 👍

I reimported your blueprint with the new functionality. But ... 😅

Message malformed: extra keys not allowed @ data['action'][1]['else'][3]['repeat']['sequence'][2]['repeat']['sequence'][8]['else'][0]['platform']

Lil' bit "tatillon" Hass 🥺

Thanks a lot

**Reply by djiwhy** [reply]
Your buymeacoffee link didn't work anymore. 😉

**Reply by etiennec78** [reply]
Thank you for noticing, I updated my link 👍

**Comment by etiennec78** [comment]
Oops, had to rush the update and didn't have the time to try it myself...
The new one should work correctly this time 😅
Thanks again for the tip, it means a lot to me !

**Reply by djiwhy** [reply]
Perfect! There are no more error messages and I was able to save the automation without any issue. I'm testing this tonight after work and I'll be back on this thread to give you feedback on how it works for me.

Please note that your blueprint is not for a garage door, but for the gate that lets me into my garden. This gate only has an opening function in its control logic, which is why I'm so interested in your blueprint and why I'm so annoying with you 😂

I'm glad to have tipped you, but your service to me is worth much more ;) 

See ya

**Reply by etiennec78** [reply]
I also use this blueprint for a gate in my garden, but mine has a pin which acts as a state toggle, so I can close and open with the same pin.

I thought you were using a pin like mine, but applied an "on" state in HA to close it too.
Since you tell me that yours cannot be closed via Home Assistant, I have pushed an update to stop your itinerary when you stop your car instead of when you activate your gate again.

Don't think you're annoying, I'm here to try to make this automation work for everyone !
Any feature request is appreciated 😉

**Reply by etiennec78** [reply]
I was thinking about your gate being stuck when a closing requests was received, and remembered that it could also receive one while leaving home if it triggered an auto-close security.
Please update to the latest version of the blueprint to avoid this issue

**Reply by djiwhy** [reply]
Well, I'm going to do that update, even though I haven't noticed this 'anomaly' yet.
I'm still fine-tuning the settings because when I arrive at my house, it takes HA about 3 minutes to open it. I'm taking it one step at a time, increasing a few parameters each time to find the right setting.
HA is already able to detect that I'm in a car and that I'm getting closer to home, but I'm still not quite sure about refining the arrival time or the distance to my destination.

I'm continuing my tests, but for the moment I have no bad behaviour to report. I just need to 'cheat' with the settings to make it more precise.

See you soon 😉👍

**Reply by etiennec78** [reply]
3 minutes seems to be a lot ! With default settings I don't have any second to wait, it is already open 5 seconds before I arrive home.
If you check on the [Vehicle Tracker Map](https://github.com/etiennec78/Home-Automation/blob/master/Automatic%20Gate/Extra/Vehicle%20tracker%20map), do you see that your phone sends a new position every 5s or less ?
Oh and by the way the distances you can set in the settings won't affect the time of arrival, it will only allow Waze to start its itinerary earlier or set an auto-closing zone if you drive away while the gate is open.
You could try tweaking the lead time, but mine in set to only 60 seconds

Thanks for your feedback :+1:
